dataproc_jupyter_plugin/controllers/dataproc.py

```diff
@@ -32,25 +32,74 @@ async def get(self):
                 client = dataproc.Client(
                     await credentials.get_cached(), self.log, client_session
                 )
-                cluster_list = await client.list_clusters(page_size, page_token)
-            self.finish(json.dumps(cluster_list))
+                runtime_list = await client.list_runtime(page_size, page_token)
+            self.finish(json.dumps(runtime_list))
         except Exception as e:
-            self.log.exception("Error fetching cluster list")
+            self.log.exception(f"Error fetching runtime template list: {str(e)}")
             self.finish({"error": str(e)})
 
 
-class RuntimeController(APIHandler):
+class ClusterListController(APIHandler):
     @tornado.web.authenticated
     async def get(self):
         try:
             page_token = self.get_argument("pageToken")
             page_size = self.get_argument("pageSize")
-            async with aiohttp.ClientSession() as client_session:
-                client = dataproc.Client(
-                    await credentials.get_cached(), self.log, client_session
-                )
-                runtime_list = await client.list_runtime(page_size, page_token)
-            self.finish(json.dumps(runtime_list))
+            client = dataproc.Client(await credentials.get_cached(), self.log)
+            cluster_list = await client.list_clusters(page_size, page_token)
+            self.finish(json.dumps(cluster_list))
         except Exception as e:
-            self.log.exception(f"Error fetching runtime template list: {str(e)}")
+            self.log.exception(f"Error fetching cluster list")
+            self.finish({"error": str(e)})
+
+
+class ClusterDetailController(APIHandler):
+    @tornado.web.authenticated
+    async def get(self):
+        try:
+            cluster = self.get_argument("cluster")
+            client = dataproc.Client(await credentials.get_cached(), self.log)
+            get_cluster = await client.get_cluster_detail(cluster)
+            self.finish(json.dumps(get_cluster))
+        except Exception as e:
+            self.log.exception(f"Error fetching get cluster")
```

Comment:
`fetching get` is redundant. Just say `Error fetching a cluster`, but also add the error itself to the log message.

e.g. ```f"Error fetching a cluster: {str(e)}"```